CBLClient/Apps/CBLTestServer-Android/app/src/main/java/com/couchbase/CouchbaseLiteServ/server/RequestHandler/PeerToPeerRequestHandler.java

```diff
@@ -0,0 +1,118 @@
+package com.couchbase.CouchbaseLiteServ.server.RequestHandler;
+
+/**
+ * Created by sridevi.saragadam on 7/9/18.
+ */
+import java.io.IOException;
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import com.couchbase.CouchbaseLiteServ.server.Args;
+import com.couchbase.lite.Database;
+import com.couchbase.lite.MessageEndpoint;
+import com.couchbase.lite.MessageEndpointConnection;
+import com.couchbase.lite.MessageEndpointDelegate;
+import com.couchbase.lite.MessageEndpointListener;
+import com.couchbase.lite.MessageEndpointListenerConfiguration;
+import com.couchbase.lite.ProtocolType;
+import com.couchbase.lite.Replicator;
+import com.couchbase.lite.ReplicatorChange;
+import com.couchbase.lite.ReplicatorChangeListener;
+import com.couchbase.lite.ReplicatorConfiguration;
+import com.couchbase.lite.URLEndpoint;
+
+public class PeerToPeerRequestHandler implements MessageEndpointDelegate{
+
+
+  public Replicator clientStart(Args args) throws Exception{
+    String ipaddress = args.get("host");
+    int port = args.get("port");
+    Database sourceDb = args.get("database");
+    String serverDBName = args.get("serverDBName");
+    String replicationType = args.get("replicationType");
+    Boolean continuous = args.get("continuous");
+    String endPointType = args.get("endPointType");
+    List<String> documentIds = args.get("documentIDs");
+    //String endPointType = "URLEndPoint";
+    ReplicatorConfiguration config;
+    Replicator replicator;
+
+    if (replicationType == null)
+    {
+      replicationType = "push_pull";
+    }
+    replicationType = replicationType.toLowerCase();
+    ReplicatorConfiguration.ReplicatorType replType;
+    if (replicationType.equals("push")) {
+      replType = ReplicatorConfiguration.ReplicatorType.PUSH;
+    } else if (replicationType.equals("pull")) {
+      replType = ReplicatorConfiguration.ReplicatorType.PULL;
+    } else {
+      replType = ReplicatorConfiguration.ReplicatorType.PUSH_AND_PULL;
+    }
+    System.out.println("serverDBName is "+ serverDBName);
+    URI uri = new URI("ws://" + ipaddress + ":"+port+"/" + serverDBName);
+    // URLEndpoint urlEndPoint= new URLEndpoint(uri);
+    // config = new ReplicatorConfiguration(sourceDb, urlEndPoint);
+    if (endPointType.toLowerCase() == "URLEndPoint"){
+
+      URLEndpoint urlEndPoint= new URLEndpoint(uri);
+      config = new ReplicatorConfiguration(sourceDb, urlEndPoint);
+    }
+    else{
+      MessageEndpoint messageEndPoint = new MessageEndpoint("p2p", uri, ProtocolType.BYTE_STREAM, this);
+      config = new ReplicatorConfiguration(sourceDb, messageEndPoint);
+    }
+    config.setReplicatorType(replType);
+    if (continuous != null) {
+      config.setContinuous(continuous);
+    }
+    else {
```

Comment:
It is good to handle this. There is no default value set either on python side or in  platform side. In the tests I have written, I made sure I passed true/false, but if any test uses and do not pass this varialbe, then this will handle